index.d.ts

```diff
@@ -546,6 +546,64 @@ console.log(stdout);
 export function execaCommand(command: string, options?: Options): ExecaChildProcess;
 export function execaCommand(command: string, options?: Options<null>): ExecaChildProcess<Buffer>;
 
+type Expression = string | number | Array<string | number>;
+
+/**
+Same as `execa()` except both file and arguments are specified in a single `command` string. For example, `execa('echo', ['unicorns'])` is the same as $\`echo unicorns\`.
+
+If the file or an argument contains spaces, they must be escaped with backslashes. This matters especially if `command` is not a constant but a variable, for example with `__dirname` or `process.cwd()`. Except for spaces, no escaping/quoting is needed.
+
+The `shell` option must be used if the `command` uses shell-specific features (for example, `&&` or `||`), as opposed to being a simple `file` followed by its `arguments`.
+
+@returns A [`child_process` instance](https://nodejs.org/api/child_process.html#child_process_class_childprocess), which is enhanced to also be a `Promise` for a result `Object` with `stdout` and `stderr` properties.
+
+@example
+```
+import {$} from 'execa';
+
+const {stdout} = await $`echo unicorns`;
+console.log(stdout);
+//=> 'unicorns'
+
+const {stdout} = await $`echo ${['unicorns', 'rainbows']}`;
+console.log(stdout);
+//=> 'unicorns rainbows'
+
+await $({stdio: 'inherit'})`echo unicorns`;
+//=> 'unicorns'
+
+const my$ = $({stdio: 'inherit'});
+await my$`echo unicorns`;
+//=> 'unicorns'
+await my$`echo rainbows`;
+//=> 'rainbows'
+```
+*/
+export const $: {
```

Comment:
We should add some type tests as well.

Reply:
Added! https://github.com/sindresorhus/execa/pull/510/commits/a287fc97c2c60ada452bc1b0b3755e596623bfac

Note: I just copied the `execaCommand` and `execaCommandSync` tests and swapped out implementations with the respective `$` APIs. Thoughts?

Reply:
This makes sense. Thanks!